frontend/src/leader-list.tsx

```diff
@@ -49,16 +53,27 @@ function colorForIndex(index: number): string {
 }
 
 export function getLeaders(pulls: Pull[], extractSigsFromPull: (pull: Pull) => Signature[]) {
-   const signatures: Signature[] = pulls.flatMap(extractSigsFromPull);
-   const users = signatures.map((signature) => signature.data.user);
-   const groupedByUsers = users
-      .reduce((grouped: Map<number, SigCount>, user: SignatureUser) => {
-         const group = grouped.get(user.id) || {user, count:0};
-         group.count++;
-         grouped.set(user.id, group)
-         return grouped;
-      }, new Map())
-      .values();
-   return Array.from(groupedByUsers)
+   const recent = toDateString(new Date(Date.now() - 86400 * 1000 * DAYS_TO_EXAMINE));
+   const recentPulls = pulls.filter((pull) => {
+      return pull.closed_at ? pull.closed_at > recent : true;
+   });
+   const sigCounts = new Map<number, SigCount>();
+   recentPulls.forEach((pull) => {
+      const sigs: Signature[] = extractSigsFromPull(pull);
+      const recentSigs = sigs.filter((sig) => sig.data.created_at > recent);
+      // Count one sig per user from this pull
+      const seenUsers = new Set<number>();
+      recentSigs.forEach((sig) => {
+         const user = sig.data.user;
+         if (seenUsers.has(user.id)) {
+            return;
+         }
+         seenUsers.add(user.id);
+         const sigCount = sigCounts.get(user.id) || {user, count:0};
+         sigCount.count++;
+         sigCounts.set(user.id, sigCount)
+      });
+   });
+   return Array.from(sigCounts.values())
```

Comment:
An alternative. Not sure if it's easier to read or not. It's shorter and uses no mutation.
Note that `countBy` and `uniq` are Lodash functions.

```suggestion
   // Generate map from userid to signatures (valid & one-per-PR)
   const sigGroups: Record<number, SignatureUser> = groupBy(recentPulls.flatMap((pull) => {
      const sigs: Signature[] = extractSigsFromPull(pull);
      const recentSigs = sigs.filter((sig) => sig.data.created_at > recent);
      // Only include each user once per PR
      return uniq(recentSigs.map(sig => sig.data.user));
   }), user => user.id);

   return Object.values(sigGroups).map((arr): SigCount => ({
      user: arr[0],
      count: arr.length,
   }));
```

Reply:
Unfortunately it seems there's a bug. Each `sig.data.user`  is `{id: number, login: string}` thus using `uniq` on an array of objects won't do anything (I think) cause they are all unique objects.

Changing that `uniq` line would probably work, but I find that harder to follow. Maybe a few more variable assignment steps would help. But to my eye it's easier to validate the correctness of the original code.

Reply:
Oh, rats. I'd originally written it with `sig.data.user.id`, which of course is much better behaved. I'm fine with the way you've got it, since you like that better.